### ISSUE TITLE
org.apache.spark package name usage

### ISSUE THREAD

**Issue by brunoborges** [mspnp/spark-monitoring#126]
Is it really required that these classes live in the `org.apache.spark` package?

My understanding is that this is an add-on to Spark.

Shouldn't _all_ classes live inside `com.microsoft.pnp` ?

1. https://github.com/mspnp/spark-monitoring/tree/master/src/spark-listeners/src/main/scala
1. https://github.com/mspnp/spark-monitoring/tree/master/src/spark-listeners-loganalytics/src/main/scala

This may cause problems to those who want/must to sign the JAR.

**Comment by hallihan** [comment]
I did a quick look and it appears the library is using private classes from org.apache.spark in several locations, so unwinding this will likely be non-trivial.  I'll create an internal workitem to track this request.

**Comment by brunoborges** [comment]
Do you know which classes are private? And are they really private?

**Comment by hallihan** [comment]
This was just a quick search-replace-build test, but here's a sampling:
```
[ERROR] ./spark-monitoring/src/spark-listeners/.../listeners/SparkListenerHandlers.scala:124: object Utils in package util cannot be accessed in package org.apache.spark.util
[ERROR] ./spark-monitoring/src/spark-listeners/.../listeners/UnifiedSparkListener.scala:43: method logEvent in trait SparkListenerEvent cannot be accessed in org.apache.spark.scheduler.SparkListenerEvent
[ERROR] ./spark-monitoring/src/spark-listeners/.../listeners/UnifiedSparkListener.scala:55: object Utils in package util cannot be accessed in package org.apache.spark.util
[ERROR] ./spark-monitoring/src/spark-listeners/.../listeners/UnifiedSparkListener.scala:68: object JsonProtocol in package util cannot be accessed in package org.apache.spark.util
[ERROR] ./spark-monitoring/src/spark-listeners/.../metrics/MetricProxies.scala:54: class RpcEndpointRef in package rpc cannot be accessed in package org.apache.spark.rpc
[ERROR] ./spark-monitoring/src/spark-listeners/.../metrics/MetricsSource.scala:8: trait Source in package source cannot be accessed in package org.apache.spark.metrics.source
[ERROR] ./spark-monitoring/src/spark-listeners/.../metrics/MetricsSourceBuilders.scala:59: value DRIVER_IDENTIFIER in object SparkContext cannot be accessed in object org.apache.spark.SparkContext
[ERROR] ./spark-monitoring/src/spark-listeners/.../metrics/MetricsSourceBuilders.scala:64: object RpcUtils in package util cannot be accessed in package org.apache.spark.util
[ERROR] ./spark-monitoring/src/spark-listeners/.../metrics/MetricsSourceBuilders.scala:64: value rpcEnv in class SparkEnv cannot be accessed in org.apache.spark.SparkEnv
[ERROR] ./spark-monitoring/src/spark-listeners/.../metrics/RpcMetricsReceiver.scala:15: trait RpcEndpoint in package rpc cannot be accessed in package org.apache.spark.rpc
[ERROR] ./spark-monitoring/src/spark-listeners/.../metrics/RpcMetricsReceiver.scala:14: trait Source in package source cannot be accessed in package org.apache.spark.metrics.source
[ERROR] ./spark-monitoring/src/spark-listeners/.../metrics/RpcMetricsReceiver.scala:16: class RpcEnv in package rpc cannot be accessed in package org.apache.spark.rpc
[ERROR] ./spark-monitoring/src/spark-listeners/.../metrics/RpcMetricsReceiver.scala:16: value rpcEnv in class SparkEnv cannot be accessed in org.apache.spark.SparkEnv
[ERROR] ./spark-monitoring/src/spark-listeners/.../metrics/UserMetricsSystem.scala:50: value DRIVER_IDENTIFIER in object SparkContext cannot be accessed in object org.apache.spark.SparkContext
[ERROR] ./spark-monitoring/src/spark-listeners/.../metrics/UserMetricsSystem.scala:66: value DRIVER_IDENTIFIER in object SparkContext cannot be accessed in object org.apache.spark.SparkContext
[ERROR] ./spark-monitoring/src/spark-listeners/.../metrics/UserMetricsSystem.scala:100: value DRIVER_IDENTIFIER in object SparkContext cannot be accessed in object org.apache.spark.SparkContext
[ERROR] ./spark-monitoring/src/spark-listeners/.../metrics/UserMetricsSystem.scala:119: value rpcEnv in class SparkEnv cannot be accessed in org.apache.spark.SparkEnv
```

**Comment by brunoborges** [comment]
@hallihan thanks Rick!

This is a good place to start.

**Comment by hallihan** [comment]
I think the signing issues should be addressed in #130 thanks to @esdhrgl1984. We still have quite a bit of package name overlap, but it appears this was part of the original design.
